### PR TITLE
Some optimisations

### PR DESCRIPTION
**Pull request by iamthesenate1**
Mostly using initialization lists; some redundant conditions

### CODE REVIEW

File: Sources/Engine/Graphics/Texture.cpp [hunk 1]
```diff
@@ -1614,12 +1614,6 @@ void ProcessScript_t( const CTFileName &inFileName) // throw char *
   INDEX i=0;
   FOREACHINLIST( CFileNameNode, cfnn_Node, FrameNamesList, it1)
   {
-    if( i != 0) {   // we have to skip first picture since it has already been done
```
Comment (ptitSeb) [inline Sources/Engine/Graphics/Texture.cpp:1617]: Why did you remove this test?

File: Sources/Engine/Rendering/SelectOnRender.cpp [hunk 2]
```diff
@@ -165,8 +165,7 @@ void SelectVertexOnRender(CBrushVertex &bvx, const PIX2D &vpix)
   // if selecting with lasso
   } else {
     // if the vertex is set in lasso buffer
-    if (_pubLassoBuffer!=NULL
-      &&_pubLassoBuffer[vpix(2)*_pixSizeI+vpix(1)]) {
```
Comment (ptitSeb) [inline Sources/Engine/Rendering/SelectOnRender.cpp:169]: Why did you remove the NULL test?

**Review by ptitSeb — COMMENTED** [review]
Some files are completely changed. Why?

File: Sources/Engine/Graphics/Texture.cpp (hunk 1 above)
Comment (iamthesenate1) [reply Sources/Engine/Graphics/Texture.cpp:1617]: Because i is set to 0 on line 1614 and as a result it will always be 0 (never !=0).

File: Sources/Engine/Rendering/SelectOnRender.cpp (hunk 2 above)
Comment (iamthesenate1) [reply Sources/Engine/Rendering/SelectOnRender.cpp:169]: Because the if from line [151](https://github.com/ptitSeb/Serious-Engine/blob/19f42367ccbb02763328d13274cfd6b92188f043/Sources/Engine/Rendering/SelectOnRender.cpp#L151) already covers the _pubLassoBuffer==NULL and the else on line 166 means that _pubLassoBuffer will always be !=NULL hence, the condition is redundant

**Comment by iamthesenate1** [comment]
> Some files are completely changed. Why?

I don't know... If I check the "Hide whitspace  changes" option in the github app everything is normal:
screen08
screen09

File: Sources/Engine/Graphics/Texture.cpp [hunk 3]
```diff
@@ -1614,12 +1614,6 @@ void ProcessScript_t( const CTFileName &inFileName) // throw char *
...
-      inPic.LoadAnyGfxFormat_t( CTString(it1->cfnn_FileName));
-      // add picture as next frame in texture
-      tex.AddFrame_t( &inPic);
-      inPic.Clear();
-    }
     i++;
```
Comment (ptitSeb) [inline Sources/Engine/Graphics/Texture.cpp:1617]: i is incremented here, so it will be different of 0 after first loop. The FOREACHINLIST macro is a `for` loop.

File: Sources/Engine/Network/EMsgBuffer.h [hunk 4]
```diff
@@ -81,12 +81,7 @@ struct TickMarker {
...
   UWORD tm_uwNumMessages;
 
-  TickMarker() {
-    tm_fTickTime = -1;
-    tm_slTickOffset = -1;
-    tm_ubAcknowledgesExpected = 0;
-    tm_uwNumMessages = 0;
-  }
+  TickMarker(): tm_fTickTime(-1), tm_fTickTime(-10), tm_ubAcknowledgesExpected(0), tm_uwNumMessages (0){}
 };
```
Comment (ptitSeb) [inline Sources/Engine/Network/EMsgBuffer.h:85]: typo here, tm_fTickTime should be `-1`, not `-10`

File: Sources/Engine/Rendering/SelectOnRender.cpp (hunk 2 above)
Comment (ptitSeb) [reply Sources/Engine/Rendering/SelectOnRender.cpp:169]: 👍

File: Sources/Engine/Graphics/Texture.cpp (hunk 3 above)
Comment (iamthesenate1) [reply Sources/Engine/Graphics/Texture.cpp]: Right. Sorry. I have reverted this change.

File: Sources/Engine/Network/EMsgBuffer.h (hunk 4 above)
Comment (iamthesenate1) [reply Sources/Engine/Network/EMsgBuffer.h:85]: Fixed. Thanks.

**Comment by ptitSeb** [comment]
Thanks :)

**Comment by iamthesenate1** [comment]
I have tried to build Serious-Engine using the latest git, but I am getting this error:
```
[  1%] Built target ecc
[  1%] Generating EntitiesMP/AirShockwave.cpp, EntitiesMP/AirShockwave.h, EntitiesMP/AirShockwave_tables.h
[  1%] Generating EntitiesMP/AmmoPack.cpp, EntitiesMP/AmmoPack.h, EntitiesMP/AmmoPack_tables.h
[  1%] Generating Engine/Classes/PlayerEntity.cpp, Engine/Classes/PlayerEntity.h, Engine/Classes/PlayerEntity_tables.h
[  1%] Generating EntitiesMP/AmmoItem.cpp, EntitiesMP/AmmoItem.h, EntitiesMP/AmmoItem_tables.h
' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AirShockwave.es(14): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AirShockwave.es(15): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AirShockwave.es(17): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AirShockwave.es(18): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoPack.es(14): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoPack.es(15): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoPack.es(16): Error: Unrecognized character '

/home/manjaro/1/Serious-Engine/Sources/Engine/Classes/PlayerEntity.es(7): Error: syntax error
make[2]: *** [CMakeFiles/ParseEntities.dir/build.make:411: Engine/Classes/PlayerEntity.cpp] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/ParseEntities.dir/build.make:433: EntitiesMP/AirShockwave.cpp] Error 1
make[2]: *** [CMakeFiles/ParseEntities.dir/build.make:455: EntitiesMP/AmmoPack.cpp] Error 1
' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoItem.es(14): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoItem.es(15): Error: Unrecognized character '

' (ASCII 0x0d)1/Serious-Engine/Sources/EntitiesMP/AmmoItem.es(16): Error: Unrecognized character '

make[2]: *** [CMakeFiles/ParseEntities.dir/build.make:444: EntitiesMP/AmmoItem.cpp] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/ParseEntities.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```
@ptitSeb Are you getting the same error? If so, can it be because of a change in this PR?

**Comment by ptitSeb** [comment]
It looks like some End of Line issue. Are you building on Mac or Windows? Try to convert the .es file to Linux (with `dos2linux` or something equivalent)

**Comment by iamthesenate1** [comment]
I'm on linux. `dos2unix` fixed the problem. Thank you very much.
Good thing it was just a local problem.

**Comment by ptitSeb** [comment]
I'll check on GitHub to have the .es files managed as the .c/.cpp files
